### DISCUSSION TITLE
problems in swarm_demo in crazy-firmware

### DISCUSSION THREAD

**Discussion by Edward-nwpu** [bitcraze/discussions#360] [Q&A] [answered]
Hi,everyone:
I'm reading the code of swarm_demos in crazy-firmware and I'm puzzled of the trajectory coefficients which the demo shows. I want to ask how to generate the coefficient from the waypoint I set? Is somebody can offer me a script to make clear this issue? Or give me enlightenment about this?
Thank you very much!

**Comment by krichardsson** [comment, marked as answer]
Hi!

You can read about the [high level trajectory format in our documentation](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/functional-areas/trajectory_formats/).

One way is to create them by hand, the compressed format is easiest to use since it is based on bezier curves.
Another option is to use the tool in [whoenig/uav_trajectories](https://github.com/whoenig/uav_trajectories) to generate polynomials from a set of waypoints.

**Reply by Edward-nwpu** [reply]
Tank you for your reply! I have read the code you suggest me to read and solved this problem and write a script by another way.
Thank you very much!
